bittensor/_axon/__init__.py

```diff
@@ -367,52 +372,71 @@ def parse_legacy_signature(
         except ValueError:
             return None
         receptor_uuid, parts = parts[-1], parts[:-1]
-        message, parts = parts[-1], parts[:-1]
-        pubkey = "".join(parts)
-        return (nonce, pubkey, message, receptor_uuid)
+        signature, parts = parts[-1], parts[:-1]
+        sender_hotkey = "".join(parts)
+        return (nonce, sender_hotkey, signature, receptor_uuid, 1)
 
-    def parse_signature(self, metadata: Dict[str, str]) -> Tuple[int, str, str, str]:
+    def parse_signature_v2(
+        self, signature: str
+    ) -> Union[Tuple[int, str, str, str, int], None]:
+        r"""Attempts to parse a signature using the v2 format"""
+        parts = signature.split(".")
+        if len(parts) != 4:
+            return None
+        try:
+            nonce = int(parts[0])
+        except ValueError:
+            return None
+        sender_hotkey = parts[1]
+        signature = parts[2]
+        receptor_uuid = parts[3]
```

Comment:
Regarding maintainability, 
what about having a separated and well tested function that give us an object with nonce, sender_hotkey, signature and receptor_uuid?

This also will help us to localize when some error occurs because of a signature not having the 4 parts (len(parts) != 4) and not mixing all the logic in the same place.

What do you think about that?

Reply:
I just realized that I was proposing something that already exists haha Sorry for that

I resolved this